tests/autopilot/autopilot_test.go

```diff
@@ -0,0 +1,458 @@
+package tests
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	apapi "github.com/libopenstorage/autopilot-api/pkg/apis/autopilot/v1alpha1"
+	. "github.com/onsi/ginkgo"
+	"github.com/onsi/ginkgo/reporters"
+	. "github.com/onsi/gomega"
+	"github.com/portworx/torpedo/drivers/node"
+	"github.com/portworx/torpedo/drivers/scheduler"
+	"github.com/portworx/torpedo/pkg/aututils"
+	. "github.com/portworx/torpedo/tests"
+)
+
+var (
+	testSuiteName            = "AUT"
+	workloadTimeout          = 5 * time.Hour
+	retryInterval            = 30 * time.Second
+	unscheduledResizeTimeout = 10 * time.Minute
+)
+
+var autopilotruleBasicTestCases = []apapi.AutopilotRule{
+	aututils.PvcRuleByUsageCapacity(50, 50, ""),
+	aututils.PvcRuleByUsageCapacity(90, 50, ""),
+	aututils.PvcRuleByUsageCapacity(50, 50, "21474836480"),
+	aututils.PvcRuleByUsageCapacity(50, 300, ""),
+}
+
+func TestAutoPilot(t *testing.T) {
+	RegisterFailHandler(Fail)
+
+	var specReporters []Reporter
+	junitReporter := reporters.NewJUnitReporter("/testresults/junit_autopilot.xml")
+	specReporters = append(specReporters, junitReporter)
+	RunSpecsWithDefaultAndCustomReporters(t, "Torpedo : Autopilot", specReporters)
+}
+
+var _ = BeforeSuite(func() {
+	InitInstance()
+})
+
+// This testsuite is used for performing basic scenarios with Autopilot rules where it
+// schedules apps and wait until workload is completed on the volumes and then validates
+// PVC sizes of the volumes
+var _ = Describe(fmt.Sprintf("{%sPvcBasic}", testSuiteName), func() {
+
+	It("has to fill up the volume completely, resize the volume, validate and teardown apps", func() {
+		var err error
+		var contexts []*scheduler.Context
+		testName := strings.ToLower(fmt.Sprintf("%sPvcBasic", testSuiteName))
+
+		Step("schedule applications", func() {
+			for i := 0; i < Inst().ScaleFactor; i++ {
+				for id, apRule := range autopilotruleBasicTestCases {
+					taskName := fmt.Sprintf("%s-%d-aprule%d", testName, i, id)
+					apRule.Name = fmt.Sprintf("%s-%d", apRule.Name, i)
+					labels := map[string]string{
+						"autopilot": apRule.Name,
+					}
+					apRule.Spec.ActionsCoolDownPeriod = int64(60)
+					context, err := Inst().S.Schedule(taskName, scheduler.ScheduleOptions{
+						AppKeys:            Inst().AppList,
+						StorageProvisioner: Inst().Provisioner,
+						AutopilotRule:      apRule,
+						Labels:             labels,
+					})
+					Expect(err).NotTo(HaveOccurred())
+					Expect(context).NotTo(BeEmpty())
+					contexts = append(contexts, context...)
+				}
+			}
+		})
+		Step("wait until workload completes on volume", func() {
+			for _, ctx := range contexts {
+				err = Inst().S.WaitForRunning(ctx, workloadTimeout, retryInterval)
```

Comment:
ValidateContext(ctx *scheduler.Context)  from common.go instead. It validates volumes as well, so you can skip the following step